pymeasure/instruments/thorlabs/thorlabspm100usb.py

```diff
@@ -23,74 +23,97 @@
-    
-    wavelength_min = Instrument.measurement("SENS:CORR:WAV? MIN", "Get minimum wavelength, in nm")
-    
-    wavelength_max = Instrument.measurement("SENS:CORR:WAV? MAX", "Get maximum wavelength, in nm")
-
+    """Represents Thorlabs PM100USB powermeter."""
 
     def __init__(self, adapter, **kwargs):
         super(ThorlabsPM100USB, self).__init__(
-            adapter, "ThorlabsPM100USB powermeter", **kwargs)
+            adapter, "ThorlabsPM100USB powermeter", **kwargs
+        )
         self.timout = 3000
-        self.sensor()
-
-    def measure_power(self, wavelength):
-        """Set wavelength in nm and get power in W
-        If wavelength is out of range it will be set to range limit"""
-        if wavelength < self.wavelength_min:
-            raise RangeException("Wavelength %.2f nm out of range: using minimum wavelength: %.2f nm" % (
-                wavelength, self.wavelength_min))
-            # explicit setting wavelenghth, althought it would be automatically set
-            wavelength = self.wavelength_min
-        if wavelength > self.wavelength_max:
-            raise RangeException("Wavelength %.2f nm out of range: using maximum wavelength: %.2f nm" % (
-                wavelength, self.wavelength_max))
-            wavelength = self.wavelength_max
-        self.wavelength = wavelength
-        return self.power
-
-    def sensor(self):
-        "Get sensor info"
-        response = self.ask("SYST:SENSOR:IDN?").split(',')
+        self._set_flags()
+
+    wavelength_min = Instrument.measurement(
+        "SENS:CORR:WAV? MIN", "Minimum wavelength, in nm"
+    )
+
+    wavelength_max = Instrument.measurement(
+        "SENS:CORR:WAV? MAX", "Maximum wavelength, in nm"
+    )
+
+    @property
+    def wavelength(self):
+        """Wavelength in nm."""
+        value = self.values("SENSE:CORR:WAV?")[0]
+        return value
+
+    @wavelength.setter
+    def wavelength(self, value):
+        """Wavelength in nm."""
+        if self.wavelength_settable:
+            value = truncated_range(value, [self.wavelength_min, self.wavelength_max])
+            self.write("SENSE:CORR:WAV {}".format(value))
+        else:
+            raise AttributeError(
+                f"{self.sensor_name} does not allow setting the wavelength."
+            )
+
+    @property
+    def power(self):
+        """Power in W."""
+        if self.is_power:
+            return self.values("MEAS:POW?")[0]
+        else:
+            raise AttributeError(f"{self.sensor_name} is not a power sensor.")
+
+    @property
+    def energy(self):
+        """Energy in J."""
+        if self.is_energy:
+            return self.values("MEAS:ENER?")[0]
+        else:
+            raise AttributeError(f"{self.sensor_name} is not an energy sensor.")
+
+    def _set_flags(self):
+        """Get sensor info and write flags."""
+        response = self.ask("SYST:SENSOR:IDN?").split(",")
         self.sensor_name = response[0]
```

Comment:
What about using `self.values` ?

Reply:
Agree, I changed it in 27bb492.

Will test soon with an actual instrument to make sure it works as expected.